### PR TITLE
[google-apps-script] Add definitions for calendar, admin directory API for advance service

### PR DESCRIPTION
**Pull request by mtgto**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - https://developers.google.com/apps-script/advanced/admin-sdk-directory
  - https://developers.google.com/apps-script/advanced/calendar
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

----

There is no JavaScript code for Google Apps Script, so I generate type definitions from GAS JSON for autocomplete.

My d.ts Generator is here.
https://github.com/mtgto/dts-google-apps-script-advanced

### CODE REVIEW

File: types/google-apps-script/google-apps-script-tests.ts [hunk 1]
```diff
@@ -35,4 +35,57 @@ const postTest = (payload: Object): string => {
     payload: payload
   };
   return UrlFetchApp.fetch(url, params).getContentText();
-};
\ No newline at end of file
+};
+
+// Calendar (Advanced service)
+declare const Calendar: GoogleAppsScript.Calendar_v3;
+const createEvent = (): void => {
```
Comment (mtgto) [inline types/google-apps-script/google-apps-script-tests.ts]: From https://developers.google.com/apps-script/advanced/calendar#creating_events

File: types/google-apps-script/google-apps-script-tests.ts [hunk 2]
```diff
@@ -35,4 +35,57 @@ const postTest = (payload: Object): string => {
...
+    // Red background. Use Calendar.Colors.get() for the full list.
+    colorId: "11"
+  };
+  event = Calendar.Events.insert(event, calendarId);
+  Logger.log('Event ID: ' + event.id);
+}
+
+// Admin Directory (Advanced service)
+declare const AdminDirectory: GoogleAppsScript.Admin_directory_v1;
+const listAllUsers = () => {
```
Comment (mtgto) [inline types/google-apps-script/google-apps-script-tests.ts]: From https://developers.google.com/admin-sdk/directory/v1/quickstart/apps-script#step_1_create_the_script

File: types/google-apps-script/apis/calendar/v3.d.ts [hunk 3]
```diff
@@ -0,0 +1,588 @@
+// Type definitions for Google Apps Script 2019-03-22
+// Project: https://developers.google.com/apps-script/
+// Definitions by: mtgto <https://github.com/mtgto/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace GoogleAppsScript {
+  namespace Calendar_v3 {
```
Comment (grant) [inline types/google-apps-script/apis/calendar/v3.d.ts]: We don't need all these namespaces.
Just GoogleAppsScript.Calendar_v3

See this example:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/google-apps-script/apis/slides/v1.d.ts#L6

File: types/google-apps-script/google-apps-script-tests.ts [hunk 4]
```diff
@@ -35,4 +35,57 @@ const postTest = (payload: Object): string => {
...
+
+// Calendar (Advanced service)
+declare const Calendar: GoogleAppsScript.Calendar_v3;
+const createEvent = (): void => {
+  const calendarId = 'primary';
+  const start = new Date();
+  const end = new Date();
+  start.setHours(10);
+  end.setHours(11);
+  let event: GoogleAppsScript.Calendar_v3.Calendar.V3.Schema.Event = {
```
Comment (grant) [inline types/google-apps-script/google-apps-script-tests.ts]:
```ts
let event: Calendar_v3.Schema.Event = {
```

**Review by grant — CHANGES_REQUESTED** [review]
Thanks!
I think there are too many namespaces. Model the types after the other libraries.
Looks good besides that.

File: types/google-apps-script/apis/calendar/v3.d.ts (hunk 3 above)
Comment (mtgto) [reply types/google-apps-script/apis/calendar/v3.d.ts]: Fix it.

https://github.com/mtgto/dts-google-apps-script-advanced/commit/ec58737259a833967b6b874132c92b46aeb5bbe3#diff-ed009b6b86b017532ef0489c77de5100R103

File: types/google-apps-script/google-apps-script-tests.ts (hunk 4 above)
Comment (mtgto) [reply types/google-apps-script/google-apps-script-tests.ts]: Fix it.

**Review by grant — APPROVED** [review]
Thanks for spending the time to create this!
LGTM.

**Pull request merged**